utils/s2n_random.c

```diff
@@ -557,7 +558,13 @@ bool s2n_supports_custom_rand(void)
 #if !defined(S2N_LIBCRYPTO_SUPPORTS_ENGINE)
     return false;
 #else
-    return s2n_libcrypto_is_openssl() && !s2n_is_in_fips_mode();
+    /* AWS-LC-FIPS supports custom rand unless s2n-tls is in FIPS mode */
+    /* OpenSSL-FIPS never supports custom rand, regardless of mode */
+    /* OpenSSL non-fips always supports custom rand */
+    /* other libcryptos never support custom rand */
+    bool awslc_fips_with_fips_enabled = s2n_libcrypto_is_awslc() && s2n_is_in_fips_mode();
```

Comment:
Is the AWS-LC check necessary? The custom RAND is never used with aws-lc anyway, right?